src/js/apps/patients/patient/flow/flow_app.js

```diff
@@ -53,6 +53,35 @@ export default SubRouterApp.extend({
       'edit': this.onEditFlow,
     });
 
+    const actions = this.flow.getActions();
+    const currentOrg = Radio.request('bootstrap', 'currentOrg');
+    const states = currentOrg.getStates();
+
+    this.listenTo(actions, {
+      'change:_state'(action) {
+        const progress = this.flow.get('_progress');
+        const previous = states.get(action._previousAttributes._state);
+        const current = states.get(action.get('_state'));
+
+        if (current.get('status') === 'done') {
+          this.flow.set({ _progress: { complete: progress.complete + 1, total: progress.total } });
+        }
+
+        if (current.get('status') !== 'done' && previous.get('status') === 'done') {
```

Comment:
so `currentId` will never equal `previousId`, because you can't "change" a state that way...what I'm checking for here is the fact that you can change from To Do to Doing back to To Do, and that'll keep decrementing because current isn't done, and it isn't previous either. That make sense?

Reply:
ah wouldn't that require both of these to be `!==`?

Reply:
don't think so? If the current state isn't done, we want to decrement. But we only want to decrement if we're coming from a done state.

Reply:
ah yeah I was thinking in terms of my original suggestion.. something like this:
```js
onChangeState(action) {
  const { complete, total } = this.flow.get('_progress');
  const current = action.get('_state');
  const previous = action.previous('_state');

  // No change in completion
  if(current !== doneStateId && previous !== doneStateId) return;

  const isNewComplete = current === doneStateId;
  this.flow.set({ _progress: { complete: complete + (isNewComplete ? 1  : -1), total } });
}
```

Reply:
ooh that's way better than what I have...